src/systems/physics/Physics.cc

```diff
@@ -660,6 +680,66 @@ void PhysicsPrivate::CreatePhysicsEntities(const EntityComponentManager &_ecm)
         auto worldPtrPhys = this->engine->ConstructWorld(world);
         this->entityWorldMap.AddEntity(_entity, worldPtrPhys);
 
+        // allow customization of contact joint surface parameters
+        auto setContactJointPropertiesCallbackFeature =
+          this->entityWorldMap.EntityCast<
+            SetContactJointPropertiesCallbackFeatureList>(_entity);
+        if (setContactJointPropertiesCallbackFeature)
+        {
+          using Policy = physics::FeaturePolicy3d;
+          using Feature = physics::SetContactJointPropertiesCallbackFeature;
+          using FeatureList = SetContactJointPropertiesCallbackFeatureList;
+          using FeatureWorld = Feature::World<Policy, FeatureList>;
+          using GCFeature = physics::GetContactsFromLastStepFeature;
+          using GCFeatureWorld = GCFeature::World<Policy, FeatureList>;
+          using ContactPoint = GCFeatureWorld::ContactPoint;
+          using ExtraContactData = GCFeature::ExtraContactDataT<Policy>;
+
+          setContactJointPropertiesCallbackFeature
+            ->AddContactJointPropertiesCallback(
```

Comment:
I've added a mechanism that keeps track of the number of customization-enabled collisions per world, and enables the callback only in case it makes sense. This removes the performance hit in 3k_shapes.sdf.